src/Dfe.EarlyYearsQualification.Web/Controllers/QuestionsController.cs

```diff
@@ -200,10 +200,8 @@ public async Task<IActionResult> WhatIsTheAwardingOrganisation(DropdownQuestionM
             return View("Dropdown", model);
         }
 
-        var cookie = userJourneyCookieService.GetUserJourneyModelFromCookie();
-        cookie.SearchCriteria = string.Empty;
-        cookie.WhatIsTheAwardingOrganisation = model.NotInTheList ? string.Empty : model.SelectedValue;
-        userJourneyCookieService.SetUserJourneyModelCookie(cookie);
+        userJourneyCookieService.SetQualificationNameSearchCriteria(string.Empty);
+        userJourneyCookieService.SetAwardingOrganisation(model.NotInTheList ? string.Empty : model.SelectedValue);
```

Comment:
Have you tested this? I think the reason I did this was because it pulls the cookie from the context and only updates the value in the cookie on the way back to the client so it looses one of the values you set when done like this. Unless the implementation has changed?

Reply:
Good point.

Maybe we can use on the fact that it's a scoped service, one service instance per request. It could read the cookie value exactly once from the response cookies, store the model as an instance variable, and write that value to the request cookies whenever a "set" method is called. Possibly more efficient, too. 🙂

Reply:
Refactored the service, and now setting multiple attributes on the cookie seems to work and has its own unit test.

Reply:
Refactored further after discussions